.github/workflows/ci_macos.yml

```
@@ -0,0 +1,137 @@
+name: CI_MacOS
+
+on:
+  push:
+  pull_request:
+  workflow_dispatch:
+jobs:
+  macos_build_and_test:
+    name: "macos_build_and_test"
+    runs-on: macos-latest
+
+    steps:
+      - name: Checkout repository
+        uses: actions/checkout@v4
+
+      - name: Debug output 
+        run: |
+          csrutil status
+          sw_vers 
+          brew install coreutils
+          sudo ln -s /usr/local/bin/gtimeout /usr/local/bin/timeout
+          sudo ln -s /opt/homebrew/bin/greadlink /usr/local/bin/greadlink
+          which timeout
+          which greadlink
+          sudo mkdir /usr/local/include
+          ls /usr/local
+          
+      - name: Install Fuse-t and dependencies
+        run: |
+          brew install python-setuptools
+          wget -q https://github.com/macos-fuse-t/fuse-t/releases/download/1.0.43/fuse-t-macos-installer-1.0.43.pkg
+          sudo installer -pkg fuse-t-macos-installer-1.0.43.pkg -target /
+          ls /usr/local/include
```

```suggestion
          wget -q https://github.com/macos-fuse-t/fuse-t/releases/download/1.0.44/fuse-t-macos-installer-1.0.44.pkg
          sudo installer -pkg fuse-t-macos-installer-1.0.44.pkg -target /
          ls /usr/local/include
```
Fix for negative lookup reply is here